versions: Use release-1.18 (commit ee9128444bec10)

fidencio opened the pull request:
Let's update CRI-O version to the commit which introduced the fix for
the "k8s-copy-file" tests.

Fixes: #1080

Signed-off-by: Fabiano Fidêncio <fidencio@redhat.com>

fidencio commented:
I'm adding the `do-not-merge` label till I also open the issue on the tests side, depending on this one, to ensure things will work as expected.

fidencio commented:
/test

fidencio commented:
/test

fidencio commented:
- **jenkins-ci-debian-10**:
```
W1110 21:37:59.580900    8068 cleanupnode.go:99] [reset] Failed to evaluate the "/var/lib/kubelet" directory. Skipping its unmount and cleanup: lstat /var/lib/kubelet: no such file or directory
W1110 21:38:09.746570    8068 cleanupnode.go:81] [reset] Failed to remove containers: output: time="2020-11-10T21:38:09Z" level=fatal msg="failed to connect: failed to connect: context deadline exceeded"
, error: exit status 1
[reset] Deleting contents of config directories: [/etc/kubernetes/manifests /etc/kubernetes/pki]
[reset] Deleting files: [/etc/kubernetes/admin.conf /etc/kubernetes/kubelet.conf /etc/kubernetes/bootstrap-kubelet.conf /etc/kubernetes/controller-manager.conf /etc/kubernetes/scheduler.conf]
[reset] Deleting contents of stateful directories: [/var/lib/dockershim /var/run/kubernetes /var/lib/cni]

The reset process does not clean CNI configuration. To do so, you must remove /etc/cni/net.d

The reset process does not reset or clean up iptables rules or IPVS tables.
If you wish to reset iptables, you must do so manually by using the "iptables" command.

If your cluster was setup to utilize IPVS, run ipvsadm --clear (or similar)
to reset your system's IPVS tables.

The reset process does not clean your kubeconfig files and you must remove them manually.
Please, check the contents of the $HOME/.kube/config file.
Failed to stop crio.service: Unit crio.service not loaded.
Failed at 29: sudo systemctl stop "${cri_runtime}"
make: *** [Makefile:148: kubernetes] Error 5
Failed at 35: sudo -E PATH="$PATH" bash -c "make kubernetes"
++ handle_error 69
++ local exit_code=2
++ local line_number=69
++ echo 'Failed at /tmp/jenkins/workspace/kata-containers-2.0-debian-PR/ci_entry_point.sh +69: .ci/jenkins_job_build.sh "${repo_to_test}"'
Failed at /tmp/jenkins/workspace/kata-containers-2.0-debian-PR/ci_entry_point.sh +69: .ci/jenkins_job_build.sh "${repo_to_test}"
++ exit 2
```

fidencio commented:
/test-centos

fidencio commented:
Okay, I'll debug it a little bit more on Thursday.

fidencio commented:
```
23:43:16 Get CRI-O sources
23:43:25 package github.com/cri-o/cri-o: no Go files in /tmp/jenkins/workspace/kata-containers-2.0-centos-PR/go/src/github.com/cri-o/cri-o
23:43:25 /tmp/jenkins/workspace/kata-containers-2.0-centos-PR/go/src/github.com/cri-o/cri-o /tmp/jenkins/workspace/kata-containers-2.0-centos-PR/go/src/github.com/kata-containers/tests
23:43:25 fatal: empty string is not a valid pathspec. please use . instead if you meant to match all paths
23:43:25 Failed at 79: git checkout "${crio_version}"
```

So, seems that for CRI-O we're only bumping to a specific version, instead of taking into consideration a random commit.
Regardless of considering a random commit being do-able (and it is!), I'd like to know whether it's a desirable change before opening an issue and working on that.
@bergwolf, @chavafg, @amshinde, @jodh-intel, any opinions on this?

fidencio commented:
/test-centos

fidencio commented:
/test-debian

fidencio commented:
@bergwolf, @chavafg, @amshinde, @jodh-intel,

Please, take a look into this approach and if it's a valid one, I'll remove the "do-not-merge" label.
Basically, I've added a tag on my own CRI-O fork, so we can easily tag a version instead of tagging a commit, which would generate more issues.  This approach would only be used till a 1.18.5 release of CRI-O is out, or if we face more regressions after that release.

What do you think?

jodh-intel commented:
Hi @fidencio - Thanks for raising. I haven't been following this I'm afraid, but I'm rather confused. Why can we not specify the commit in the versions database? We've done this a lot in the past and if you really want to map back to the version, you can just do a `git describe --tags $commit` in the git clone for the particular project in question. I'd prefer we don't use personal repos for tracking things like this too tbh.

fidencio commented:
> Hi @fidencio - Thanks for raising. I haven't been following this I'm afraid, but I'm rather confused. Why can we not specify the commit in the versions database? We've done this a lot in the past and if you really want to map back to the version, you can just do a `git describe --tags $commit` in the git clone for the particular project in question. I'd prefer we don't use personal repos for tracking things like this too tbh.

@jodh-intel, at first I've removed the version and tried to use the `commit: $commit_hash` and it just broke.  Indeed, using `git describe --tags $commit` is something that didn't cross my mind and may be the simplest way to solve the issue.  Thanks **a lot** for the suggestion. I've updated the PR and will check if it works as expected.

fidencio commented:
/test-centos

fidencio commented:
/test

fidencio commented:
/test

devimc reviewed (approved):
thanks @fidencio
